blog/_posts/2021-10-30-debuging-root-scripts-in-eclipse.md

```diff
@@ -0,0 +1,227 @@
+---
+title: "Debugging CERN ROOT scripts and ROOT-based programs in Eclipse IDE"
+layout: archive
+author: Petr Stepanov
+---
+
+<center>
+{% include figure_image
+   img="/assets/images/setup-eclipse-ide-with-cern-root.jpg"
+   style="width: 300px"
+%}
+</center>
+
+CERN ROOT is a rather complex object-oriented framework. It is written in C++, a language with complete manual control over the memory. Therefore, execution of your your ROOT script may sometimes lead to a crash providing minimal information in the stack trace. CERN ROOT is just a framework and it does not provide any out-of-the-box solutions for debugging ROOT scripts. Hence, a question about debugging ROOT scripts now and then arises in the ROOT community. 
+
+In this blog post, I will share my experience with debugging CERN ROOT scripts and ROOT-based programs. We will utilize Eclipse CDT (C/C++ Development Tooling) Integrated Desktop Environment (IDE), a free software. Eclipse coupled with the GNU Debugger (GDB) provides enhanced code development infrastructure that includes code highlighting, easy navigation between C++ sources and headers, debugging your script as well as ROOT source code, and many more.
+
+A few words about the operating system (OS). In this post, I will use Linux-based OS as a system of choice. The same process applies to macOS, but may require a code signing procedure. On Windows, a minGW installation is a prerequisite.
```

Comment:
Will a minGW installation help with Eclipse-debugging of MSVC-compiled ROOT? If not, maybe just remove the Windows part, or say "Windows is a different story"? :-)